returnn/tf/layers/rec.py

```diff
@@ -5856,7 +5861,7 @@ def __init__(self, num_heads, total_key_dim,
       prev_mask = None
       if self._rec_previous_layer:
         assert self.input_data.time_dim_axis is None
-        assert attention_left_only
+        assert attention_mask == "only_left"
```

Comment:
```suggestion
        assert attention_mask == "left_only"
```
?

Reply:
Here, `attention_mask == "left_only" or callable(attention_mask)` would need to be supported too if we want to allow setting a callable `attention_mask` inside a rec layer.
Would the provided custom `attention_mask` then need to handle masking away the future, or should `SelfAttentionLayer` always apply such masking on top of the provided mask?
I think I'd slightly prefer the latter.

Reply:
Maybe `attention_mask` can be made only of type callable (or layer) and we leave the previous behavior in place?
This way the mask can be anything and be simple, then the length or past-only restrictions are enforced as before.
What do you think?

Reply:
Yes, I think that's a good idea :+1: